### ISSUE TITLE
CalDAV or iCal Support

### ISSUE THREAD

**Issue by JohnDoe257** [wekan/wekan#2836]
I would like to know what the state of CalDAV or iCal Support in Wekan is. I have seen some threads regarding this topic, but all of them were in any form linked to https://github.com/wekan/wekan/issues/808 which was closed after implementing the calendar view.

Is this feature still planned in any form or would this require a new feature request by now?

Best regards

**Comment by GitGramm** [comment]
Maybe this helps?
https://github.com/LukasGasior1/wekan-ical-server

**Comment by xet7** [comment]
@GitGramm 

Does ical servers in general have any kind of authentication? It would be nice to have some authentication.

It seems that the Python code just goes through Wekan calendar events to make read-only calendar page. That kind of code could as well be written in Javascript directly in Wekan code.

**Comment by xet7** [comment]
Other way is, if you don't care about authentication, is to have setting at board to have public calendar URL enabled, similarly like setting board public.

**Comment by GitGramm** [comment]
For me read-only and no authentication is ok.

AFAIK there is a way to put authentication to an ical/caldav calender.
At least sabre https://sabre.io/dav/authentication/ can handle some authentication?

I am not the developer of the wekan-ical-server script but had the requirement to implement it on my family-board and it is now running since a year without problems:-)

I am able to code a litte bit in javascript but i am not sure if I am skilled enough to contribute that code-conversion to wekan in production-ready quality...
Maybe for someone more experienced its easy to switch the 45 codelines into a Javascript for Wekan?

**Comment by makayabou** [comment]
I don't know for everyone, but for me, caldav integration would be like:
1/ you create a task in an another front-end for caldav server (e.g nextcloud)
2/ a card is created in wekan board linked to caldav server

and otherway around:
1/ I create a card in wekan
2/ It creates a task in caldav server

**Comment by peterk** [comment]
This feature could probably be divided into three types:

1. A calendar feed per board which, if enabled, provides a URL for a read-only calendar feed. Anyone with the URL can subscribe to the feed in their calendar application and it will include all tasks with either an expiry date or tasks that have both the started and ended dates set.
2. A calendar feed per user per board. If enabled it is also openly available but if a user is removed from the board or from Wekan entirely it will no longer work. This would enable people to have certain tasks visible in their calendar application of choice.
3. A calendar feed per user that includes all the tasks the user is assigned to (or member of?) for all boards in the system.

**Comment by stlehmann** [comment]
FYI I created a [fork of wekan-ical-server](https://github.com/stlehmann/wekan-ical-server) that supports Python 3 and runs the server as a Docker container.

**Comment by xet7** [comment]
@stlehmann 

Thanks! I forked your repo to https://github.com/wekan/wekan-ical-server .

**Comment by xet7** [comment]
I also added links to Time Tracking wiki page:
https://github.com/wekan/wekan/wiki/Time-Tracking

**Comment by lucker999** [comment]
> 
> 
> This feature could probably be divided into three types:
> 
>     1. A calendar feed per board which, if enabled, provides a URL for a read-only calendar feed. Anyone with the URL can subscribe to the feed in their calendar application and it will include all tasks with either an expiry date or tasks that have both the started and ended dates set.
> 
>     2. A calendar feed per user per board. If enabled it is also openly available but if a user is removed from the board or from Wekan entirely it will no longer work. This would enable people to have certain tasks visible in their calendar application of choice.
> 
>     3. A calendar feed per user that includes all the tasks the user is assigned to (or member of?) for all boards in the system.

I would definitely like to have a possibility to manage tasks (my or assigned to me) with another todo app on mobile device. See, this is a perfectly useful and very practical use case: I have a todo list on my phone (or tablet) for a week, which is instantly at hand-reach. I can see current tasks, filter it by boards (projects), priority, started/not started (backlog / planned lsists), mark it progress (if finished - task card moved to "Done" list, for example). To acheve this it needs a CalDAV support since other open-source apps use this protocol (and to avoid Google and such). 

For example, nextcloud deck can be synced with Thunderbird (CalDAV), but for now it lacks sync-back capabilities and many other features to manage tasks in another ToDo list app properly. 

I hope (very much) that in Wekan it could be done. Care to consider?

**Comment by ViOffice** [comment]
FYI, until this feature is properly implemented upstream, we took the idea of the `wekan-ical-server` python script and implemented a similar service in PHP. This solution adds the benefit of not having to store user-credentials in plain text (however API auth tokens are stored in a database).

Addittionally, this may be used by an indefinite amount of users of the same wekan-instance, without manual intervention by the server host. Similarly to `wekan-ical-server`

, the output is a simple `.ical` file which can also be used as webcal sync (read-only). Authentication is not yet implemented, however each user gets their own unique 16-digits ID, which is relatively hard to guess.

Some feedback is welcome, since everyone in this thread is probably interested in some sort of solution (although having this feature directly in wekan would be much better).

- [Main repo on Codeberg](https://codeberg.org/ViOffice/wekan-ical-php)
- [Github-Mirror](https://github.com/ViOffice/wekan-ical-php)

![wekan-ical-php_02](https://user-images.githubusercontent.com/78919040/121820099-26e81400-cc91-11eb-9a96-8d91f1785bab.jpg)
![wekan-ical-php_01](https://user-images.githubusercontent.com/78919040/121820101-2780aa80-cc91-11eb-8a54-9b3431f5e924.jpg)

**Comment by xet7** [comment]
@ViOffice 

Thanks a lot! I forked that repo to:
https://github.com/wekan/wekan-ical-php

**Comment by ViOffice** [comment]
@xet7 Thanks to you too! Keep in mind, that this project may receive additional features and hopefully a bunch of improvements in the coming week. If you want, I can message you about it, so you can keep the fork in sync :)

**Comment by xet7** [comment]
@ViOffice

Yes you can message me when there is updates, at GitHub issues or to support@wekan.team 

Since you seem to know PHP, what do you think about my prototype at https://github.com/wekan/php ?

**Comment by ViOffice** [comment]
> Since you seem to know PHP, what do you think about my prototype at https://github.com/wekan/php ?

Just started picking up PHP, but I'll look into it and hack along. Maybe I can contribute :) Thanks!

**Comment by xet7** [comment]
@ViOffice 

Thanks a lot ! Yes, currently that PHP prototype has only small amount of code, it is much easier to contribute to than to Meteor Wekan.
